tests/test_functional.py

```diff
@@ -0,0 +1,176 @@
+import asyncio
+
+import pytest
+
+from aviary.core import ToolCall, ToolRequestMessage, fenv
+from aviary.functional import FunctionalEnvironment
+
+
+# Fixtures
+@pytest.fixture
+def basic_env():
+    @fenv.start()
+    def test_env(param: str):
+        return f"Test {param}", {"param": param}
+
+    @test_env.tool()
+    def test_tool(x: str, state) -> str:  # noqa: ARG001
+        """Test tool that processes input."""
+        return f"Processed {x}"
+
+    return test_env
+
+
+@pytest.fixture
+def async_env():
+    @fenv.start()
+    async def test_env(param: str):
+        await asyncio.sleep(0.1)
+        return f"Async Test {param}", {"param": param}
+
+    @test_env.tool()
+    async def test_tool(x: str, state) -> str:  # noqa: ARG001
+        """Test tool that processes input."""
+        await asyncio.sleep(0.1)
+        return f"Async Processed {x}"
+
+    return test_env
+
+
+def test_environment_creation(basic_env):
+    env = basic_env("test")
+    assert isinstance(env, FunctionalEnvironment)
+    assert len(env.tools) == 1
+    assert env.tools[0].info.name == "test_tool"
+
+
+@pytest.mark.asyncio
+async def test_sync_env_reset(basic_env):
+    env = basic_env("test_param")
+    obs, tools = await env.reset()
+
+    assert len(obs) == 1
+    assert obs[0].content == "Test test_param"
+    assert obs[0].role == "user"
+    assert len(tools) == 1
+
+
+@pytest.mark.asyncio
+async def test_async_env_reset(async_env):
+    env = async_env("test_param")
+    obs, tools = await env.reset()
+
+    assert len(obs) == 1
+    assert obs[0].content == "Async Test test_param"
+    assert obs[0].role == "user"
+    assert len(tools) == 1
+
+
+# Test State Management
+@pytest.mark.asyncio
+async def test_state_management():
+    @fenv.start()
+    def test_env():
+        return "Test", {"custom_value": 42}
+
+    @test_env.tool()
+    def modify_state(state) -> None:
+        """Modify the state."""
+        state.reward = 1.0
+        state.done = True
+        state.extras["new_value"] = 100
+
+    env = test_env()
+    await env.reset()
+
+    assert env.state.reward == 0
+    assert env.state.done is False
+    assert env.state.extras["custom_value"] == 42
+
+    tool_call = ToolCall.from_name("modify_state")
+    action = ToolRequestMessage(tool_calls=[tool_call])
+    msgs, reward, done, truncated = await env.step(action)
+
+    assert reward == 1.0
+    assert done is True
+    assert env.state.extras["new_value"] == 100
+
+
+@pytest.mark.asyncio
+async def test_invalid_start_function():
+    @fenv.start()
+    def invalid_env():
+        return "Only one value"  # Should return tuple
+
+    env = invalid_env()
+    with pytest.raises(TypeError):
+        await env.reset()
+
+
+async def test_invalid_state_dict():
+    @fenv.start()
+    def invalid_env():
+        return "Test", "not_a_dict"  # Should be a dict
```

Comment:
This confuses me, why is it not a dict, but the comments says should be a dict. It sort of feels like a TODO comment, but I don't think it is

Can you update the comment to be a bit clearer